src/components/BigNumberPad.js

```diff
@@ -32,74 +32,68 @@ const padNumbers = [
     ['.', '0', '<'],
 ];
 
-class BigNumberPad extends React.PureComponent {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            timer: null,
-        };
-    }
+function BigNumberPad(props) {
+    const [timer, setTimer] = useState(null);
 
     /**
      * Handle long press key on number pad.
      * Only handles the '<' key and starts the continuous input timer.
      *
      * @param {String} key
      */
-    handleLongPress(key) {
-        // Only handles deleting.
+    const handleLongPress = (key) => {
         if (key !== '<') {
             return;
         }
-        this.props.longPressHandlerStateChanged(true);
-        const timer = setInterval(() => {
-            this.props.numberPressed(key);
+
+        props.longPressHandlerStateChanged(true);
+
+        const newTimer = setInterval(() => {
+            props.numberPressed(key);
```

Comment:
When changed from class to functional component, the native device's long press stopped working. It is just deleting the last text only; numberPressed was passing the old value, and due to that, the selection value doesn't have the the correct value, and the previous text is just deleted. More context here:  https://github.com/Expensify/App/issues/39079#issuecomment-2023717810